src/main/java/com/amazon/jenkins/ec2fleet/EC2FleetCloud.java

```diff
@@ -487,9 +494,18 @@ private void addNewSlave(final AmazonEC2 ec2, final String instanceId) throws Ex
         // Check if we have the address to use. Nodes don't get it immediately.
         if (address == null) return; // Wait some more...
 
+        int numExecutors;
+        if (scaleExecutorsByWeight) {
+            Double instanceTypeWeight = stats.getInstanceTypeWeight(instance.getInstanceType());
+            Double instanceWeight = Math.ceil(this.numExecutors * instanceTypeWeight);
```

Comment:
```instanceTypeWeight``` could be ```< 1```, based on current implementation it will be always one executor, which is fine, but could you please add notes here about that case. Thx.